test/e2eprovider/types/types.go

```diff
@@ -0,0 +1,24 @@
+package e2eprovider
+
+// KeyVaultObject holds keyvault object related config
+type KeyVaultObject struct {
+	// the name of the Azure Key Vault objects
+	ObjectName string `json:"objectName" yaml:"objectName"`
+	// the filename the object will be written to
+	ObjectAlias string `json:"objectAlias" yaml:"objectAlias"`
+	// the version of the Azure Key Vault objects
+	ObjectVersion string `json:"objectVersion" yaml:"objectVersion"`
+	// the type of the Azure Key Vault objects
+	ObjectType string `json:"objectType" yaml:"objectType"`
+	// the format of the Azure Key Vault objects
+	// supported formats are PEM, PFX
+	ObjectFormat string `json:"objectFormat" yaml:"objectFormat"`
+	// The encoding of the object in KeyVault
```

Comment:
These are very specific to a single provider. We should simplify this to just name, version and return a static value that the driver can deterministically validate.

Reply:
done